[13.0][FIX] stock_reserve_rule: allow removal strategy based on lots

kirca opened the pull request:
The lot is ignored in _apply_strategy which makes it not possible to
create removal rules based on it. A practical example is to create a
rule with a quant_domain on lot_id.use_date (Best Before Date). Even
though the quants will be filtered, the lots wont be taken into
consideration in _apply_strategy and so random lots will be reserved.
With the the signature for the _apply_strategy_%s methods is changed
to require returning the lot also or set it to False if it is not used.

rousseldenis commented:
/ocabot rebase

rousseldenis commented:
@kirca Thanks for this.

Regarding the changes, IMHO we should add the owner_id too as it is a criteria too.

kirca committed:
[FIX] stock_reserve_rule: allow removal strategy based on lots and/or owner

The lot is ignored in _apply_strategy which makes it not possible to
create removal rules based on it. A practical example is to create a
rule with a quant_domain on lot_id.use_date (Best Before Date). Even
though the quants will be filtered, the lots wont be taken into
consideration in _apply_strategy and so random lots will be reserved.
With tis the signature for the _apply_strategy_%s methods is changed
to require returning the lot also or set it to None if it is not used.

The same rationale applies for the owner of the quant.

kirca commented:
Thank you @rousseldenis . I've added the owner too and did the rebase.

rousseldenis commented:
@kirca Great. Maybe test cases for lots and owner would be great !

kirca commented:
Hello @rousseldenis , finally there is a test case now as well :)

rousseldenis commented:
@sebalix @jcadhoc Maybe a review here ?

elvise commented:
@kirca can you rebase ?
(runboat issue)

francesco-ooops commented:
@rousseldenis can you try to launch rebase?

rousseldenis commented:
/ocabot rebase

rousseldenis commented:
@francesco-ooops As this one is really old, you can maybe do another PR if needed.

francesco-ooops commented:
@rousseldenis thanks, we will!

kirca commented:
@rousseldenis @francesco-ooops , sorry for the delayed response. I just did the rebase

francesco-ooops reviewed (approved):
Functional is ok!

@rousseldenis shall we merge?

rousseldenis commented:
/ocabot merge minor
